### PR TITLE
Numeric lib should handle strings more gracefully

### PR DESCRIPTION
**Pull request by happythenewsad**
## WHAT
Numeric lib should handle string input more gracefully.

## WHY
Redis values are string by default.

### Screenshots
(If applicable. Also, please censor any sensitive data)

### Notion Card Links
none

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  yes
Have you deployed to Staging? | not yet
Self-Review: Have you done an initial self-review of the code below on Github? | na
Spec Review: Have you reviewed the spec and ensured this meets requirements and/or matches design mockups? | (N/A

### CODE REVIEW

File: services/QuillLMS/app/lib/utils/numeric.rb [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 module Utils::Numeric
   def self.to_human_string(numeric)
-    raise ArgumentError, "Not a numeric" unless numeric.is_a?(Numeric)
+    return numeric.to_s unless numeric.is_a?(Numeric)
```
Comment (anathomical) [inline services/QuillLMS/app/lib/utils/numeric.rb]: If something can be converted to a number, like these values in Redis, don't we actually want the utility of the `Numeric` conversion here?  Like, if we pull `"5700000"` out of Redis, don't we want this to actually return "6 million" or whatever?

**Review by anathomical — CHANGES_REQUESTED** [review]
Do we actually want to just throw strings back?  Shouldn't we actually process numbers that just happen to be stored as strings?

File: services/QuillLMS/app/lib/utils/numeric.rb (hunk 1 above)
Comment (happythenewsad) [reply services/QuillLMS/app/lib/utils/numeric.rb]: Yes, that's a nice enhancement; I've added it. I leveraged the fact the `to_f` will never raise an exception based on it's input. 

`"asdf".to_f => 0.0`

**Review by anathomical — COMMENTED** [review]
Nice.  Do we need to deal with the possibility of someone passing `nil` or `"hello"` into the function mistakenly, or should we just deal with that if/when it happens?

**Comment by brendanshean** [comment]
> Nice. Do we need to deal with the possibility of someone passing `nil` or `"hello"` into the function mistakenly, or should we just deal with that if/when it happens?

I'm guessing this probably related: I pushed a temporary [fix](https://github.com/empirical-org/Empirical-Core/pull/9332) due to this [error](https://one.newrelic.com/nr1-core/errors/overview/MjYzOTExM3xBUE18QVBQTElDQVRJT058NTQ4ODU2ODc1?account=2639113&duration=10800000&state=e3a4cfcf-c418-926a-1d72-f2718b48341d)

**Comment by happythenewsad** [comment]
> 

Both of these cases are currently handled. This helper function is only called in one place in the code, FYI.

**Review by anathomical — APPROVED** [review]

**Comment by happythenewsad** [comment]
> > Nice. Do we need to deal with the possibility of someone passing `nil` or `"hello"` into the function mistakenly, or should we just deal with that if/when it happens?
> 
> I'm guessing this probably related: I pushed a temporary [fix](https://github.com/empirical-org/Empirical-Core/pull/9332) due to this [error](https://one.newrelic.com/nr1-core/errors/overview/MjYzOTExM3xBUE18QVBQTElDQVRJT058NTQ4ODU2ODc1?account=2639113&duration=10800000&state=e3a4cfcf-c418-926a-1d72-f2718b48341d)

That's right - you can revert that commit once this one deploys.

**Pull request merged**